makefiles/Makefile.devhub-content-new

```diff
@@ -0,0 +1,59 @@
+GIT_BRANCH=$(shell git rev-parse --abbrev-ref HEAD)
+USER=$(shell whoami)
+
+STAGING_URL="http://docs-devhub-staging.s3-website-us-east-1.amazonaws.com"
+STAGING_BUCKET=docs-devhub-staging
+
+STRAPI_URL=http://54.219.137.111:1337
+
+PRODUCTION_BUCKET=developer-hub-prod
+PRODUCTION_URL=https://mongodb.com/developer
+
+PROJECT=devhub
+MUT_PREFIX ?= $(PROJECT)
+REPO_DIR=$(shell pwd)
+
+COMMIT_HASH=$(shell git rev-parse --short HEAD)
+
+GITHUB_USER = $(shell printenv GITHUB_BOT_USERNAME)
+GITHUB_PASS = $(shell printenv GITHUB_BOT_PASSWORD)
+SNOOTY_DB_USR = $(shell printenv MONGO_ATLAS_USERNAME)
+SNOOTY_DB_PWD = $(shell printenv MONGO_ATLAS_PASSWORD)
+
+include ~/shared.mk
+
+update-snooty:
+	cd ../../snooty-devhub; \
+	git fetch --all; \
+	git reset --hard origin/master; \
+	npm install --production
+
+get-build-dependencies:
+	@curl https://raw.githubusercontent.com/mongodb/docs-worker-pool/meta/publishedbranches/devhub-content.yaml > ${REPO_DIR}/published-branches.yaml
+
+next-gen-html: update-snooty
+	# snooty parse and then build-front-end
+	@echo ${SNOOTY_DB_PWD} | snooty build "${REPO_DIR}" "mongodb+srv://${SNOOTY_DB_USR}:@cluster0-ylwlz.mongodb.net/snooty?retryWrites=true" --commit "${COMMIT_HASH}"; \
+	if [ $$? -eq 1 ]; then \
+		exit 1; \
+	else \
+		exit 0; \
+	fi
+	rsync -az --exclude '.git' ${REPO_DIR}/../../snooty-devhub ${REPO_DIR}; 
+	cp ${REPO_DIR}/.env.production ${REPO_DIR}/snooty-devhub; 
+	curl https://raw.githubusercontent.com/mongodb/devhub/master/gatsby-config.prod-new.js > ${REPO_DIR}/snooty/gatsby-config.js
```

Comment:
```suggestion
	curl https://raw.githubusercontent.com/mongodb/devhub/master/gatsby-config.prod-new.js > ${REPO_DIR}/snooty-devhub/gatsby-config.js
```

Reply:
We'll need to make sure it curls to the `snooty-devhub` dir for this approach, but I have another concern in that these job runs aren't atomic - after this gets executed, we'd overwrite the `gatsby-config.js` for every following job, until the docker container gets rebuilt.

@GuruPKK correct me if I'm wrong 🤔

Reply:
Oh interesting, curious what we do now to distinguish between gatsby-config and gatsby-config.prod?

Reply:
> Oh interesting, curious what we do now to distinguish between gatsby-config and gatsby-config.prod?

We could be extra safe here by changing line 42 to 
```
rsync -az --exclude '.git' ${REPO_DIR}/../../snooty-devhub/ ${REPO_DIR}/snooty-devhub-tmp;
```
and then changing further references to `snooty-devhub` after line 42  to `snooty-devhub-tmp`, so we're extra confident that everything is happening in a new directory, and only this job run is occurring in that directory.

Reply:
@jestapinski cassidy's suggestion needs to be applied otherwise we will be overwriting the config

Reply:
Yeah, in general I think we'd be much more confident in this if we rsync to a directory specific to this job run.

This would also help us during any debugging of a job run that needs to happen - and would ensure that we don't run into issues if this makefile runs in parallel with the base devhub job run.

Reply:
Yeah it will be better to keep it in different dir, and curl to that specific dir. this will have the changes local to devhub and doesnt impact other builds if we turn on multi processing in auto builder